File: src/main/java/com/sber/yamlprocessor/model/Reverse.java
package com.sber.yamlprocessor.model;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;

import java.util.List;

public class Reverse {
    
    @Valid
    @NotNull(message = "Status is required")
    private B3Status status;
    
    @Valid
    @NotNull(message = "Output is required")
    private List<ReverseOutput> output;

    public Reverse() {}

    public Reverse(B3Status status, List<ReverseOutput> output) {
        this.status = status;
        this.output = output;
    }

    // Getters and Setters
    public B3Status getStatus() {
        return status;
    }

    public void setStatus(B3Status status) {
        this.status = status;
    }

    public List<ReverseOutput> getOutput() {
        return output;
    }

    public void setOutput(List<ReverseOutput> output) {
        this.output = output;
    }
}

File: src/main/kotlin/com/sber/yamlprocessor/model/Reverse.kt
package com.sber.yamlprocessor.model

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.OrderColumn
import jakarta.persistence.Table

@Entity
@Table(name = "reverse_entry")
class Reverse(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @JsonIgnore
    @Column(name = "id")
    var dbId: Long? = null,

    @ManyToOne(optional = false)
    @JoinColumn(name = "result_id")
    @JsonIgnore
    var result: Result? = null,

    @ManyToOne(optional = false, fetch = FetchType.EAGER)
    @JoinColumn(name = "b3_status", referencedColumnName = "code", nullable = false)
    var status: B3StatusDictionary = B3StatusDictionary(code = "INITIATED"),

    @OneToMany(mappedBy = "reverse", cascade = [CascadeType.ALL], orphanRemoval = true)
    @OrderColumn(name = "output_sort")
    var output: MutableList<ReverseOutput> = mutableListOf()
)
